Fill.py

```diff
@@ -15,8 +17,81 @@ class FillError(RuntimeError):
     pass
 
 
-def _log_fill_progress(name: str, placed: int, total_items: int) -> None:
-    logging.info(f"Current fill step ({name}) at {placed}/{total_items} items placed.")
+class FillLogger():
+    """
+        FillLogger class variables:
+        min_size -- minimum number of items we care about reporting
+        min_time -- minimum time (in seconds) between reports (used in dynamic logging only)
+        step     -- number of items between reports (used in fixed logging only)
+    """
+    min_size: int = 1000
+    min_time: float = 0.25
+    step: int = 1000
```

Comment:
If you do it like this, then IDEs like PyCharm and VSCode/VSCodium will show this documentation in a mouse-over pop-up when you put your mouse over these names in other places.

```suggestion
    min_size: int = 1000
    """ minimum number of items we care about reporting """
    min_time: float = 0.25
    """ minimum time (in seconds) between reports (used in dynamic logging only) """
    step: int = 1000
    """ number of items between reports (used in fixed logging only) """
```

Reply:
Thanks; since I don't use IDEs I was unaware. Updated.